src/main/java/com/taskadapter/redmineapi/AttachmentManager.java

```diff
@@ -110,7 +110,41 @@ public Attachment uploadAttachment(String contentType, File content)
         }
     }
 
+/**
+	 * Uploads an attachment.
+	 * 
+	 * @param fileName
+	 *            file name of the attachment.
+	 * @param contentType
+	 *            content type of the attachment.
+	 * @param content
+	 *            attachment content stream.
+	 * @return attachment content.
+	 * @throws RedmineException
+	 *             if something goes wrong.
+	 * @throws IOException
+	 *             if input cannot be read. This exception cannot be thrown yet (I am not sure if http client can distinguish "network" errors and local errors) but is will be good to distinguish
+	 *             reading errors and transport errors.
+	 */
+	public Attachment uploadAttachment(File file, String contentType) throws RedmineException, IOException {
+		final String token;
+		try {
+			token = transport.upload(file);
+			final Attachment result = AttachmentFactory.create();
+			result.setToken(token);
+			result.setContentType(contentType);
+			result.setFileName(file.getName());
+			return result;
+		} catch (RedmineException e) {
+			unwrapException(e, "uploadStream");
+			throw e;
+		}
+	}
+
     /**
+     * 
+     * @decrepated
```

Comment:
Have u a solution to change a InputStream to a File ? You can purpose changes too !
I think it's a decrepated Function because she works only with small file attachments..